server.py

```
@@ -0,0 +1,255 @@
+        )
+    ]
+)
+
+# --- WebSocket Helper ---
+async def send_to_client(websocket, message_dict):
+    """Sends a JSON message to the client."""
+    try:
+        await websocket.send(json.dumps(message_dict))
+    except websockets.exceptions.ConnectionClosed:
+        logger.warning("Client connection closed, unable to send message.")
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)